sdk/core/azure-core/src/http/request.cpp

```diff
@@ -58,7 +58,18 @@ std::string Request::getEncodedUrl()
   auto queryString = std::string("");
   for (auto pair : queryParameters)
   {
-    queryString += (queryString.empty() ? "?" : "&") + pair.first + "=" + pair.second;
+    if (pair.first == "comp")
+    {
+      if (!queryString.empty())
+      {
+        queryString[0] = '&';
```

Comment:
It seems like having a "PrependQueryParameter" function or something like that would be a good idea here...

Reply:
The order of query parameters should not matter. "comp=" should not have to be the 1st query parameter.